torch/distributed/pipelining/_schedule_visualizer.py

```diff
@@ -0,0 +1,176 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+
+"""
+This visualizer requires matplotlib to be installed.
+"""
+
+from typing import Optional, Union
+from unittest import mock
```

Comment:
what does this dependency do? Would this add a new dependency to pytorch lib?

Reply:
unittest is part of python's stdlib so it wont cause an issue!